packages/core/src/utils.ts

```diff
@@ -378,6 +380,39 @@ export const fetchSphinxManagedBaseUrl = () => {
     : 'https://www.sphinx.dev'
 }
 
+export const readSphinxLock = async (): Promise<SphinxLock> => {
+  if (!existsSync(SPHINX_LOCK_PATH)) {
+    throw new Error(
+      'Missing `sphinx.lock` file, run `npx sphinx sync` to regenerate this file. We recommend committing it to version control.'
+    )
+  }
```

Comment:
This error can only be thrown when running the `deploy` command which just reads the local `sphinx.lock` file. When the `propose` command is run, we always fetch the latest `sphinx.lock` file, so this error cannot occur.